src/tests/uat/recommendation_uat.py

```diff
@@ -0,0 +1,423 @@
+    result = compare(context, new_rule.signed_by, raw_data, 'signed_by') and result
+    if new_rule.application_:
+        result = compare_application(f"{context}.application", new_rule.application_,
+                                     raw_data['application']) and result
+    else:
+        result = check_not_present(context, raw_data, 'application') and result
+    return result
+
+
+def compare_workflow(context, workflow, raw_data):
+    """
+    Compare a RecommendationWorkflow object against the data that should represent it.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        workflow (RecommendationWorkflow): The RecommendationWorkflow object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, workflow.changed_by, raw_data, 'changed_by')
+    result = compare(context, workflow.create_time, raw_data, 'create_time') and result
+    result = compare(context, workflow.ref_id, raw_data, 'ref_id') and result
+    result = compare(context, workflow.status, raw_data, 'status') and result
+    result = compare(context, workflow.update_time, raw_data, 'update_time') and result
+    result = compare(context, workflow.comment, raw_data, 'comment') and result
+    return result
+
+
+def compare_recommendation(context, recommendation, raw_data):
+    """
+    Compare a Recommendation object against the data that should represent it.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        recommendation (Recommendation): The Recommendation object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, recommendation.policy_id, raw_data, 'policy_id')
+    result = compare(context, recommendation.recommendation_id, raw_data, 'recommendation_id') and result
+    result = compare(context, recommendation.rule_type, raw_data, 'rule_type') and result
+    if recommendation.impact_:
+        result = compare_impact(f"{context}.impact", recommendation.impact_, raw_data['impact']) and result
+    else:
+        result = check_not_present(context, raw_data, 'impact') and result
+    if recommendation.new_rule_:
+        result = compare_new_rule(f"{context}.new_rule", recommendation.new_rule_, raw_data['new_rule']) and result
+    else:
+        result = check_not_present(context, raw_data, 'new_rule') and result
+    if recommendation.workflow_:
+        result = compare_workflow(f"{context}.workflow", recommendation.workflow_, raw_data['workflow']) and result
+    else:
+        result = check_not_present(context, raw_data, 'workflow') and result
+    return result
```

Comment:
Can you print a message if 'result' is false before returning it?

Reply:
The `compare()` function does that.

```
 if not result:
        print(f"field value {raw_field} did not match in {context} - object value {object_field}, "
              f"raw value {raw_data.get(raw_field, None)}")
```